src/reactapp/src/components/paypalExpress/hooks/usePaypalExpress.js

```diff
@@ -0,0 +1,138 @@
+import LocalStorage from '../../../utils/localStorage';
+import { config } from '../../../config';
+import usePaymentMethodCartContext from '../../paymentMethod/hooks/usePaymentMethodCartContext';
+import useBillingAddressCartContext from '../../billingAddress/hooks/useBillingAddressCartContext';
+
+export default function usePaypalExpress() {
+  const [, appActions] = useContext(AppContext);
+  const { setErrorMessage, setPageLoader } = appActions;
+  const { selectedMethod } = useShippingMethodCartContext();
+  const { cartBillingAddress } = useBillingAddressCartContext();
+  const query = window.location.href.split('?')[1];
+  const [processPaymentEnable, setProcessPaymentEnable] = useState(false);
+  const { selectedPaymentMethod, methodList } = usePaymentMethodCartContext();
+  const cartId = LocalStorage.getCartId();
+
+  /*
+   Check if is possible to proceed on placing the order.
+   */
+  useEffect(() => {
+    if (
+      query &&
+      selectedMethod?.methodCode &&
+      (selectedPaymentMethod?.code === '' ||
+        selectedPaymentMethod?.code === 'paypal_express')
+    )
+      setProcessPaymentEnable(true);
+  }, [query, setProcessPaymentEnable, selectedMethod, selectedPaymentMethod]);
+
+  useEffect(() => {
+    const processPaypalExpress = async () => {
+      setProcessPaymentEnable(false);
+      /*
+       Get paypal express required data from response query
+       */
+      const { token, PayerID: payerId } = getTokenPayerId(query);
+      const paymentMethod = _objToArray(methodList).find(
+        method => method.code === 'paypal_express'
+      );
+
+      if (
+        !token ||
+        !payerId ||
+        !cartId ||
+        !selectedMethod ||
+        !paymentMethod ||
+        !cartBillingAddress
+      ) {
+        return;
+      }
+
+      /*
+       Set payment method on cart and place the order
+       */
+      try {
+        setPageLoader(true);
+        await setPaymentMethodPaypalExpress({ payerId, token });
+        const response = await placeOrder();
+        if (response && response.order_number) {
+          LocalStorage.clearCheckoutStorage();
+          window.location.replace(
+            `${config.baseUrl}/checkout/onepage/success/`
+          );
+        }
+      } catch (error) {
+        setPageLoader(false);
+        setErrorMessage(
+          __(
+            'Something went wrong while adding the payment method to the quote.'
+          )
+        );
+      }
+    };
+    if (processPaymentEnable) {
+      processPaypalExpress(cartId, methodList);
+    }
+  }, [
+    cartId,
+    methodList,
+    processPaymentEnable,
+    query,
+    selectedMethod,
+    setErrorMessage,
+    setPageLoader,
+    cartBillingAddress,
+  ]);
+
+  /*
+   Get the customer token from the BE and redirect to paypal.
+   */
+  const authorizeUser = useCallback(async () => {
+    if (!selectedMethod?.carrierCode || !cartBillingAddress?.firstname) {
```

Comment:
Try to use `_get` function as i mentioned above please

Reply:
Done